src/cli/generate-idlnames.js

```diff
@@ -339,15 +339,15 @@ async function saveIdlNamesFragments(names, folder) {
   }
 
   await createFolderIfNeeded(folder);
-  await Promise.all(Object.values(names).map(idl => {
+  await Promise.all(Object.entries(names).map(([name, idl]) => {
     const res = [];
     if (idl.defined) {
       res.push(serializeNode(idl.defined));
     }
     if (idl.extended) {
       idl.extended.map(node => res.push(serializeNode(node)));
     }
-    const filename = path.join(folder, idl.name + '.idl');
+    const filename = path.join(folder, name + '.idl');
```

Comment:
I'm not sure I understand how this works - why won't undefined no longer appear?

Reply:
The previous code assumed that the `name` property was always set. That is not the case when the object only contains partial definitions. FYI, the code that generates these objects is at:
https://github.com/w3c/reffy/blob/master/src/cli/generate-idlnames.js#L158.

The fix is to use the key to the object in `names` instead, which is always set to the interface name.

Same logic as the one used for the `idlnamesparsed` folder:
https://github.com/w3c/reffy/blob/master/src/cli/generate-idlnames.js#L316